packages/x-data-grid-premium/src/components/GridFooterCell.tsx

```diff
@@ -10,10 +11,10 @@ const GridFooterCellRoot = styled('div', {
   name: 'MuiDataGrid',
   slot: 'FooterCell',
   overridesResolver: (_, styles) => styles.footerCell,
-})<{ ownerState: OwnerState }>(({ theme }) => ({
-  fontWeight: theme.typography.fontWeightMedium,
-  color: (theme.vars || theme).palette.primary.dark,
-}));
+})<{ ownerState: OwnerState }>({
+  fontWeight: vars.typography.fontWeight.medium,
+  color: vars.colors.foreground.accent,
+});
```

Comment:
I'm confused. This is Material UI API, we have to stick to using Material UI on the Material UI version, why are we changing this. Can we revert?

Reply:
> the Data Grid version for Material UI needs to follow the Material UI API and conventions so the data grid deeply integrates with it. So this doesn't work in that case?

So do you think we should keep relying on `@mui/system`? That means we'd still be shipping emotion/styled even for non-mui users like shadcn.

> we could have demo with all the slots styled, userland, no style in the npm package.

So this means we'd be writing each style once for each styling solution, right? Once with material-ui, once with tailwind/shadcn, etc.

Reply:
Splitting styles in a separate package is not viable, I have outlined the arguments against it here: https://www.notion.so/mui-org/DataGrid-styling-plan-19acbfe7b6608059bb6dc31608fab53a